pkg/reconciler/v1alpha1/config/gc.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"strconv"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	RevisionGcConfigName = "config-gc"
+)
+
+type RevisionGc struct {
+	// Delay duration after a revision create before considering it for GC
+	StaleRevisionCreateDelay time.Duration
+	// Timeout since a revision lastPinned before it should be GC'd
+	StaleRevisionTimeout time.Duration
+	// Minimum number of generations of revisions to keep before considering for GC
+	StaleRevisionMinimumGenerations int64
+	// Minimum staleness duration before updating lastPinned
+	StaleRevisionLastpinnedDebounce time.Duration
+}
+
+func NewRevisionGcFromConfigMap(configMap *corev1.ConfigMap) (*RevisionGc, error) {
+	c := RevisionGc{}
+
+	for _, dur := range []struct {
+		key          string
+		field        *time.Duration
+		defaultValue time.Duration
+	}{{
+		key:          "stale-revision-create-delay",
+		field:        &c.StaleRevisionCreateDelay,
+		defaultValue: 5 * time.Minute,
+	}, {
+		key:          "stale-revision-timeout",
+		field:        &c.StaleRevisionTimeout,
+		defaultValue: 5 * time.Minute,
+	}, {
+		key:          "stale-revision-lastpinned-debounce",
+		field:        &c.StaleRevisionLastpinnedDebounce,
+		defaultValue: 1 * time.Minute,
```

Comment:
Given the changes to our default ResyncPeriod, we should probably make these much longer.